### PR TITLE
Add support for --literal flag and literal entry in config

### PR DESCRIPTION
**Pull request by PanGan21**
Add support for `--literal` in order to opt out showing filenames with quotes
Closes: #894

---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests
- [x] Add changelog entry
- [x] Update default config/theme in README (if applicable)
- [x] Update man page at lsd/doc/lsd.md (if applicable)

### CODE REVIEW

File: doc/lsd.md [hunk 1]
```diff
@@ -140,6 +140,9 @@ lsd is a ls command with a lot of pretty colours and some other stuff to enrich
 `--header`
 : Display block headers
 
+`--literal`
```
Comment (zwpaper) [inline doc/lsd.md:143]: according to https://man7.org/linux/man-pages/man1/ls.1.html

we can add `-N` as short name

File: doc/lsd.md [hunk 2]
```diff
@@ -140,6 +140,9 @@ lsd is a ls command with a lot of pretty colours and some other stuff to enrich
...
+: Do not show quotes on filenames
```
Comment (zwpaper) [inline doc/lsd.md:144]: > print entry names without quoting

ref: https://man7.org/linux/man-pages/man1/ls.1.html

File: src/app.rs [hunk 3]
```diff
@@ -181,6 +181,10 @@ pub struct Cli {
     #[arg(long, hide = !cfg!(windows))]
     pub system_protected: bool,
 
+    /// When showing files do not quote their names
+    #[arg(long)]
+    pub literal: bool,
```
Comment (zwpaper) [inline src/app.rs:186]: `-N` and description align with https://man7.org/linux/man-pages/man1/ls.1.html

File: src/core.rs [hunk 4]
```diff
@@ -71,7 +73,7 @@ impl Core {
             // or require a raw output (like the `wc` command).
             inner_flags.layout = Layout::OneLine;
 
-            flags.should_quote = false;
```
Comment (zwpaper) [inline src/core.rs:74]: how about changing `should_quote` to `literal`?

**Review by zwpaper — CHANGES_REQUESTED** [review]
thanks for the contribution!

it mostly looks great! only with some nit we can discuss

File: doc/lsd.md (hunk 1 above)
Comment (PanGan21) [reply doc/lsd.md]: Should be fixed now!

File: doc/lsd.md (hunk 2 above)
Comment (PanGan21) [reply doc/lsd.md]: Fixed it as well according to the manual

File: src/app.rs (hunk 3 above)
Comment (PanGan21) [reply src/app.rs]: Fixed it!

File: src/core.rs (hunk 4 above)
Comment (PanGan21) [reply src/core.rs:74]: Changed!

File: src/meta/name.rs [hunk 5]
```diff
@@ -153,21 +153,21 @@ impl Name {
                 format!(
                     "{}{}",
                     icons.get(self),
-                    self.hyperlink(self.escape(self.file_name(), quote), hyperlink)
+                    self.hyperlink(self.escape(self.file_name(), !quote), hyperlink)
```
Comment (PanGan21) [inline src/meta/name.rs:156]: Let me know if you think `quote` arg should change to literal as well in order to avoid this negation here

Comment (zwpaper) [reply src/meta/name.rs:156]: Yes! it would be great if you can also update this one!

Comment (PanGan21) [reply src/meta/name.rs]: Fixed it as well!

**Comment by zwpaper** [comment]
hi @PanGan21, can you fix the CI errors

**Comment by PanGan21** [comment]
> hi @PanGan21, can you fix the CI errors

I think I am confused a bit with the errors of the ci. Could they be caused because I did `cargo update` and updated the `cargo.lock` file?
I did that because when building the project locally I was getting the following error:
`error[E0635]: unknown feature `proc_macro_span_shrink` which was fixed after updating cargo versions

**Comment by PanGan21** [comment]
> hi @PanGan21, can you fix the CI errors

I was using wrong rustc version, so I had to update `Cargo.lock`. Now I reverted it and used the correct rust version.
Could you run the ci one more time to check if everything is fixed now?

**Comment by zwpaper** [comment]
thanks for the contribution! @PanGan21

File: CHANGELOG.md [hunk 6]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [v1.0.0] - 2023-08-25
 
 ### Added
+- Add support for `--literal` from [PanGan21](https://github.com/PanGan21)
```
Comment (shilangyu) [inline CHANGELOG.md:17]: Should this have gone to an already-released v1.0.0?

Comment (zwpaper) [reply CHANGELOG.md:17]: good catch! actually, I was planning to use the GitHub auto-generate release note, this file may not be necessary, can you help create a PR to fix it and add a notice that we are no longer using this file, we use the GitHub auth-generate one instead.

Comment (shilangyu) [reply CHANGELOG.md:17]: Not exactly sure what you meant, but opened one here: https://github.com/lsd-rs/lsd/pull/907

Let me know if it is what you meant.

**Pull request merged**